File: src/main/java/com/github/sorusclient/client/adapter/ITextStyle.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter;

public interface ITextStyle {
    ITextClickEvent getClickEvent();
    ITextHoverEvent getHoverEvent();
    String getInsertion();
    TextFormatting getColor();
    boolean isBold();
    boolean isItalic();
    boolean isObfuscate();
    boolean isStrikethrough();
    boolean isUnderline();
}

File: src/main/kotlin/com/github/sorusclient/client/adapter/ITextStyle.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter

interface ITextStyle {
    val clickEvent: ITextClickEvent?
    val hoverEvent: ITextHoverEvent?
    val insertion: String?
    val color: TextFormatting?
    val bold: Boolean
    val italic: Boolean
    val obfuscated: Boolean
    val strikethrough: Boolean
    val underlined: Boolean
}